x-pack/plugins/cases/server/client/metrics/all_cases/aggregations/status_counts.ts

```diff
@@ -0,0 +1,57 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { CasePersistedStatus } from '../../../../common/types/case';
+import { caseStatuses } from '../../../../../common/types/domain';
+import { CASE_SAVED_OBJECT } from '../../../../../common/constants';
+import type { CasesMetricsResponse } from '../../../../../common/types/api';
+import type { AggregationBuilder, AggregationResponse } from '../../types';
+
+export class StatusCounts implements AggregationBuilder<CasesMetricsResponse> {
+  build() {
+    return {
+      status: {
+        terms: {
+          field: `${CASE_SAVED_OBJECT}.attributes.status`,
+          size: caseStatuses.length,
+        },
+      },
+    };
+  }
+
+  formatResponse(aggregations: AggregationResponse) {
+    const aggs = aggregations as StatusAggregate;
+    const buckets = aggs?.status?.buckets ?? [];
+
+    const open = buckets.find(({ key }) => key === CasePersistedStatus.OPEN);
```

Comment:
what about running through buckets just once? No idea about "buckets" size

Reply:
Good point, even if they are only three good to be more generic. Done in [`69447f9` (#201098)](https://github.com/elastic/kibana/pull/201098/commits/69447f9065a127cdb5e40ea970a5ece4a3315ec4).